nbgitpuller/handlers.py

```diff
@@ -38,6 +49,47 @@ def emit(self, data):
         self.write('data: {}\n\n'.format(serialized_data))
         yield self.flush()
 
+    def setup_plugins(self, provider):
+        pm = pluggy.PluginManager("nbgitpuller")
+        pm.add_hookspecs(hookspecs)
+        num_loaded =pm.load_setuptools_entrypoints("nbgitpuller", name=provider)
+        if num_loaded == 0:
+            raise ProviderException(f"The provider key you supplied in the URL could not be found: {provider}")
+        return pm
+
+    @gen.coroutine
+    def progress_loop(self, queue):
+        """
+        The loop below constantly checks the queue paremeter for messages
+        that are being sent to the UI so the user is kept aware of progress related to
+        the downloading of archives and the merging of files into the user's home folder
+
+        :param queue: This is either the download_queue or the original pull queue
+        """
+        while True:
+            try:
+                progress = queue.get_nowait()
+            except Empty:
+                yield gen.sleep(0.1)
+                continue
+            if progress is None:
+                yield gen.sleep(5)
```

Comment:
Separate question about `gen.sleep(5)`.

I'm not following the logic of what it will mean to sleep 5 here.
- Does it mean that a progressbar could lag 5 seconds before updating again?
- Does it mean that the user may be redirected up to 5 slower than needed?

I want to make sure we don't have something as sluggish as 5 seconds which seems like an infinity in user experience contexts.

Reply:
it means it will wait 5 seconds. I can shorten it; I would wait if you want the user to see the output when everything goes right. 2 seconds?

Reply:
I don't understand the situation, but I'm expecting there wouldn't be a need to have a arbitrary sleep duration that isn't very short unless we are trying to mitigate something we can't know about and then we simply opt to have a reasonable default about that unknown.

If that is the case, perhaps whats missing is simply an inline comment to clarify that "we wait five seconds here as we can't know about ... so we estimate five seconds is about right".

Or ah, is this a "We have displayed all the messages, but before we redirect the user, we wait 5 seconds so the user has time to read about it!" situation?

If that is the case, then I would think it's preferable to not wait at all unless we expect it to be a warning message or error message.

Reply:
I don't understand the code well enough yet, so my questions are a bit confused =/

Reply:
Yes!:
> Or ah, is this a "We have displayed all the messages, but before we redirect the user, we wait 5 seconds so the user has time to read about it!" situation?

OK, I will remove the wait.

Reply:
Ah!!!